multicluster/test/e2e/service_test.go

```diff
@@ -50,31 +40,53 @@ func (data *MCTestData) testServiceExport(t *testing.T) {
 			t.Fatalf("Error when creating Service %s in cluster %s: %v", clusterServiceName, clusterName, err)
 		}
 	}
-	// Create Pod and Service in west cluster
+	// Create Pod and Service in west and ease cluster
 	createPodAndService(westCluster, westClusterTestService)
-	defer deletePodWrapper(t, data, westCluster, multiClusterTestNamespace, podName)
-	defer deleteServiceWrapper(t, testData, westCluster, multiClusterTestNamespace, westClusterTestService)
-
-	// Create Pod and Service in east cluster
 	createPodAndService(eastCluster, eastClusterTestService)
-	defer deletePodWrapper(t, data, eastCluster, multiClusterTestNamespace, podName)
-	defer deleteServiceWrapper(t, testData, eastCluster, multiClusterTestNamespace, eastClusterTestService)
+	return podName
+}
 
-	deployServiceExport := func(clusterName string) {
-		if err := data.deployServiceExport(clusterName); err != nil {
-			t.Fatalf("Error when deploy ServiceExport in cluster %s: %v", clusterName, err)
-		}
+func (data *MCTestData) tearDownTestResources(t *testing.T, podName string) {
+	deleteServiceWrapper(t, testData, westCluster, multiClusterTestNamespace, westClusterTestService)
+	deleteServiceWrapper(t, testData, eastCluster, multiClusterTestNamespace, eastClusterTestService)
+	deletePodWrapper(t, data, westCluster, multiClusterTestNamespace, podName)
+	deletePodWrapper(t, data, eastCluster, multiClusterTestNamespace, podName)
+}
+
+// Deploy service exports in east and west clusters
+func setUpServiceExport(data *MCTestData, t *testing.T) {
+	if err := data.deployServiceExport(westCluster); err != nil {
+		t.Fatalf("Error when deploy ServiceExport in west cluster: %v", err)
+	}
+	if err := data.deployServiceExport(eastCluster); err != nil {
+		t.Fatalf("Error when deploy ServiceExport in east cluster: %v", err)
 	}
+	time.Sleep(importServiceDelay)
+}
 
-	// Deploy ServiceExport in west cluster
-	deployServiceExport(westCluster)
-	defer data.deleteServiceExport(westCluster)
+func tearDownServiceExport(data *MCTestData) {
+	data.deleteServiceExport(westCluster)
+	data.deleteServiceExport(eastCluster)
+}
 
-	// Deploy ServiceExport in east cluster
-	deployServiceExport(eastCluster)
-	defer data.deleteServiceExport(eastCluster)
-	time.Sleep(importServiceDelay)
+// Try to curl the counter part services in east and west clusters.
+// If we get status code 200, it means that the resources is exported by the east cluster
+// and imported by the west cluster.
+func testProbeMCService(t *testing.T, data *MCTestData) {
+	data.testProbeMCService(t)
+}
 
+func testANP(t *testing.T, data *MCTestData) {
+	data.testANP(t)
+}
+
+func (data *MCTestData) testProbeMCService(t *testing.T) {
+	data.probeMCServiceFromCluster(t, eastCluster, westClusterTestService)
+	data.probeMCServiceFromCluster(t, westCluster, eastClusterTestService)
+}
+
+func (data *MCTestData) testANP(t *testing.T) {
```

Comment:
This's not correct, you wrapped Service connectivity related test into `testANP`, only the last line `data.verifyMCServiceACNP(t, gwClientName, eastIP)` is for ANP.

Reply:
Good catch. Changed.